invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/UserDetails/components/ItemsSearch/ItemsSearch.js

```diff
@@ -37,23 +38,25 @@ export default class ItemsSearch extends Component {
   };
 
   executeSearch = queryString => {
+    queryString = queryString || this.props.queryString;
+    this.setState({ prevSearchQuery: queryString });
+
     if (queryString) {
       return this.fetchItems(queryString);
-    } else {
-      return this.fetchItems(this.props.queryString);
     }
   };
 
   _onPasteHandler = e => {
     let queryString = e.clipboardData.getData('Text');
     this.executeSearch(queryString).then(data => {
-      if (this.props.items.hits[0].status === 'LOANABLE') {
-        this.checkoutItem(this.props.items.hits[0], this.props.patron).then(
-          () => {
-            this.clearResults();
-            this.fetchUpdatedCurrentLoans(this.props.patron);
-          }
-        );
+      const { patron } = this.props;
+      const { hits } = this.props.items;
+      if (!_isEmpty(hits) && hits[0].metadata.status === 'LOANABLE') {
```

Comment:
actually now if think about it, we should make sure that it returns only one hit and if it returns more raise an error, or at least do not check out automatically, I know it is not likely as the barcodes are unique, but maybe it is worth it to check anyway. WDYT?

Reply:
Can we do anything to guarantee that barcodes are unique in the database?

If we can't then yes I agree. Perhaps we can do a checkout if there is only one hit and show the search results if there are multiple hits?

Reply:
yes I would go with that.
I checked, and we are not currently ensuring the unique constraint on the barcode, I've created an issue to solve it.